yt_dlp/extractor/doodstream.py

```diff
@@ -40,18 +41,43 @@ class DoodStreamIE(InfoExtractor):
             'description': 'Stacy Cruz Cute ALLWAYSWELL | DoodStream.com',
             'thumbnail': 'https://img.doodcdn.com/snaps/8edqd5nppkac3x8u.jpg',
         }
+    }, {
+        'url': 'https://dood.so/d/wlihoael8uog',
+        'md5': '5144b8066c68c5a5a3321eb623ab7272',
+        'info_dict': {
+            'id': 'wlihoael8uog',
+            'ext': 'mp4',
+            'title': 'VID 20220319 161659',
+            'description': 'VID 20220319 161659',
+            'thumbnail': 'https://img.doodcdn.com/snaps/8edqd5nppkac3x8u.jpg',
+        }
     }]
 
     def _real_extract(self, url):
+        parts = compat_urlparse.urlsplit(url)
+        hostname = parts.hostname
+        path = parts.path
         video_id = self._match_id(url)
-        url = f'https://dood.to/e/{video_id}'
+
+        title = None
+        description = None
+        if path.startswith('/d/'):
+            webpage = self._download_webpage(url, video_id)
+            iframe = self._html_search_regex(r'<iframe +src="([^"]+)"', webpage, 'iframe')
+            url = f'https://{hostname}{iframe}'
+            title = self._html_search_meta(['og:title', 'twitter:title'], webpage, default=None)
+            description = self._html_search_meta(
+                ['og:description', 'description', 'twitter:description'], webpage, default=None)
```

Comment:
duplicate code. To avoid this, it would be better to do
```py
webpage = original_webpage = self._download_webpage(...)
if path.startswith(...):
    url = 'https://%s%s' % (hostname, self._html_search_regex(...))
    webpage = self._download_webpage(..., note='Downloading embedded page')
...
return {
    'title': self._html_search_meta(..., original_webpage, ...)
    ...
}
```